ibm/service/secretsmanager/resource_ibm_sm_private_certificate_configuration_action_set_signed.go

```diff
@@ -0,0 +1,95 @@
+package secretsmanager
+
+import (
+	"context"
+	"fmt"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM/secrets-manager-go-sdk/v2/secretsmanagerv2"
+)
+
+func ResourceIbmSmPrivateCertificateConfigurationActionSetSigned() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIbmSmPrivateCertificateConfigurationActionSetSignedCreateOrUpdate,
+		ReadContext:   resourceIbmSmPrivateCertificateConfigurationActionSetSignedRead,
+		UpdateContext: resourceIbmSmPrivateCertificateConfigurationActionSetSignedCreateOrUpdate,
+		DeleteContext: resourceIbmSmPrivateCertificateConfigurationActionSetSignedDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name that uniquely identifies a configuration",
+			},
+			"certificate": &schema.Schema{
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+					if removeNewLineFromCertificate(oldValue) == removeNewLineFromCertificate(newValue) {
+						return true
+					}
+					return false
+				},
+				ForceNew:    true,
+				Description: "The PEM-encoded certificate.",
+			},
+		},
+	}
+}
+
+func resourceIbmSmPrivateCertificateConfigurationActionSetSignedCreateOrUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	secretsManagerClient, err := meta.(conns.ClientSession).SecretsManagerV2()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	region := getRegion(secretsManagerClient, d)
+	instanceId := d.Get("instance_id").(string)
+	secretsManagerClient = getClientWithInstanceEndpoint(secretsManagerClient, instanceId, region, getEndpointType(secretsManagerClient, d))
+
+	createConfigurationActionOptions := &secretsmanagerv2.CreateConfigurationActionOptions{}
+
+	configurationActionPrototypeModel, err := resourceIbmSmPrivateCertificateConfigurationActionSetSignedPrototype(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	createConfigurationActionOptions.SetConfigActionPrototype(configurationActionPrototypeModel)
+	createConfigurationActionOptions.SetName(d.Get("name").(string))
+
+	_, response, err := secretsManagerClient.CreateConfigurationActionWithContext(context, createConfigurationActionOptions)
+	if err != nil {
+		log.Printf("[DEBUG] CreateConfigurationActionWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("CreateConfigurationActionWithContext failed %s\n%s", err, response))
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s/set_signed", region, instanceId, d.Get("name").(string)))
```

Comment:
I dont see the use of having `set_signed` in the id. As each action has its respective resource.
If you still want it, it should be fine.

Reply:
It's for separating the action's resource from the configuration's resource.